tiledb/sm/array/array.cc

```diff
@@ -346,39 +340,38 @@ Status Array::get_max_buffer_size(
         Status::ArrayError("Cannot get max buffer size; "
                            "Array was not opened in read mode"));
 
-  // Check if attribute is null
-  if (attribute == nullptr)
-    return LOG_STATUS(
-        Status::ArrayError("Cannot get max buffer size; Attribute is null"));
-
-  RETURN_NOT_OK(compute_max_buffer_sizes(subarray));
+  // Check if name is null
+  if (name == nullptr)
+    return LOG_STATUS(Status::ArrayError(
+        "Cannot get max buffer size; Attribute/Dimension name is null"));
 
-  // Normalize attribute name
-  std::string norm_attribute;
-  RETURN_NOT_OK(
-      ArraySchema::attribute_name_normalized(attribute, &norm_attribute));
+  // Check if name is attribute or dimension
+  bool is_dim = array_schema_->is_dim(name);
+  bool is_attr = array_schema_->is_attr(name);
 
-  // Check if attribute exists
-  auto it = last_max_buffer_sizes_.find(norm_attribute);
-  if (it == last_max_buffer_sizes_.end())
+  // Check if attribute/dimension exists
+  if (name != constants::coords && !is_dim && !is_attr)
     return LOG_STATUS(Status::ArrayError(
-        std::string("Cannot get max buffer size; Attribute '") +
-        norm_attribute + "' does not exist"));
+        std::string("Cannot get max buffer size; Attribute/Dimension '") +
+        name + "' does not exist"));
 
-  // Check if attribute is fixed sized
-  if (array_schema_->var_size(norm_attribute))
+  // Check if attribute/dimension is fixed sized
+  if (array_schema_->var_size(name))
     return LOG_STATUS(Status::ArrayError(
-        std::string("Cannot get max buffer size; Attribute '") +
-        norm_attribute + "' is var-sized"));
+        std::string("Cannot get max buffer size; Attribute/Dimension '") +
+        name + "' is var-sized"));
+
+  RETURN_NOT_OK(compute_max_buffer_sizes(subarray));
 
   // Retrieve buffer size
+  auto it = last_max_buffer_sizes_.find(name);
```

Comment:
We previously had a check that this was not `!= last_max_buffer_sizes_.end()`. If the expectation is that this will never happen, how about an assert() here?

Reply:
done